### ISSUE TITLE
Docs: no-duplicate-imports - confusing option description and examples

### ISSUE THREAD

**Issue by aubergine10** [eslint/eslint#6455]
I'm working through the es6 docs (#5446) to make them more consistent and ran in to this:

http://eslint.org/docs/rules/no-duplicate-imports#options

I've not used ES6 `import` and `export` so I might be missing some pre-requisite knowledge, but the option description and examples are highly confusing to me. I get parse errors when pasting the code in to the online demo, so I can't even tell which line of code would generate the rule violation.

Could someone with ES6 `import` / `export` knowledge help with improved description and examples?

As a start I think it might be beneficial to split the first option example in to two code blocks - or at least add a comment to denote to separate examples in the one block (I assume they are two separate examples due to the 3 blank lines in middle of example, or is it just a single example with extraneous line feeds?):

```js
// example 1: <reason why this fails>
import { merge } from 'module';
import path from 'another-module'; // do we need this line?

export { find } from 'module';

// example 2: <reason why this fails>
import _ from 'module';
const find = _.find;

export { find as lodashFind } from 'module';
```

cc doc authors: @SimenB, @scriptdaemon

**Comment by aubergine10** [comment]
Also, I feel this rule doc would benefit greatly from an opening description that explains what problem the rule aims to prevent.

**Comment by platinumazure** [comment]
In the online demo, try adding this comment:

`/* eslint-env es6 */`

That will hopefully get the parser to work and you can verify the examples.

**Comment by aubergine10** [comment]
Yup, tried that (it seems the option is enabled by default along with all other rules in the checkbox panel at bottom of that page) but initially tried it with the inline comment. It looks like either the code viewer or whatever parses the JS is breaking due to the syntax rather than any specific rules selected.

Here's what I pasted in to the demo:

```js
/* eslint-env es6 */
/*eslint no-duplicate-imports: ["error", { includeExports: true }]*/

import { merge } from 'module';
import path from 'another-module';

export { find } from 'module';


import _ from 'module';
const find = _.find;

export { find as lodashFind } from 'module';
```

(Taken direct from current doc page, with addition of `es6` env setting).

The error displayed in the demo is:

> 4:1 - Parsing error: 'import' and 'export' may appear only with 'sourceType: module' (null)

**Comment by pedrottimark** [comment]
Ouch, been there, done that. Our pain becomes others’ gain. Trouble shooting a bit:
* The **Espree** demo parses the code correctly with ECMA Version: 6 and Source Type: module
* Unfortunately the **ESLint** demo has different options and it displays that error message even when I select the modules check box under ECMA Features. At least a limitation, probably an error.
* As far as I know, there is no eslint comment to specify `"sourceType": "module"`

What I did was set up a project with configuration according to link below, copied example code to separate files, and then linted to verify expected errors for incorrect or no errors for correct either via command line or editor integration. For example, I use Atom editor with linter and linter-eslint community packages. The double-ouch was I set it and forgot it, and then later got confused by parsing errors in non-strict example code because modules have implicit strict mode.

If you keep a list of rule docs under **ECMAScript 6** where this happens, you can submit a future pull request to fix this issue by adding a sentence (to be determined) which includes the following link:

http://eslint.org/docs/user-guide/configuring#specifying-parser-options

That is an example of a chunk of content for the guidelines issue after the pull request is merged.

**Comment by pedrottimark** [comment]
Yes, I agree on the main point see if you can work with the authors to bring the contrast between examples of incorrect and correct code into sharper focus.

Reading quickly, I came to the mistaken conclusion that the problem in the default examples is importing `merge` 2 times, but coming back from the demos and reading slower the second time, I see the problem is multiple `import` statements which refer to the same module.

If I understand correctly, the problem in the examples for the `"includeExports": false` option is exporting `find` 2 times. There the confusion is because the incorrect and correct examples do not have minimum differences for a clear contrast what is the error.

This reminds me to go ahead and add some guidelines for contrast in pairs of examples :)

Looking at it from the viewpoint of the original contributors (who I trust will read this issue as constructive :) there is a difficult balance. Unless the incorrect code is a bit messy, the software quality problem is not plausible. It needs to look realistic enough so readers can match it with a mistake they made and never want to make again but simple enough for people who are new to ES6 features not to miss the forest for the trees.

**Comment by aubergine10** [comment]
> This reminds me to go ahead and add some guidelines for contrast in pairs of examples :)

Yes, just going through the es6 rules I also noticed this was a common problem.

**Comment by SimenB** [comment]
Hey there!

I agree the phrasing is clunky, I remember thinking so when I wrote it. The only doc comment I got in the PR was to not import from `lodash`, but from something more generic.

I'll open a PR with alternate phrasing, and see if we can't make sense of it all! 😄

**Comment by SimenB** [comment]
See #6463
